### DISCUSSION TITLE
setValues not working for react-select

### DISCUSSION THREAD

**Discussion by niralparmar** [react-hook-form/react-hook-form#1842] [Q&A] [answered]
i am trying to set value onchange of react select, since react select returns object rather value, i need to get value and set it to my object
my expected output is 
`{ tag: { member : ['1','2']}}`
but all i am getting on submit it
`{ tag: { }}`
i tried but no luck
`methods.setValue("tag.member", tags);`
` methods.setValue("tag", { member: tags });`

Here is my [codesandbox](https://codesandbox.io/s/musing-glitter-0brjd)

appreciate any help

**Comment by bluebill1049** [comment]
Take a look at this example: https://codesandbox.io/s/react-hook-form-controller-079xx which have react-select and try to work on top of it and compare the diff.

**Comment by niralparmar** [comment, marked as answer]
@bluebill1049 thanks for the reply, in that example there is no explicit  setValue for react-select.
without setValue here what data looks like. as per my requirement  member should be array of string rather object
`{
  "tag": {
    "member": [
      {
        "value": "chocolate",
        "label": "Chocolate"
      },
      {
        "value": "strawberry",
        "label": "Strawberry"
      }
    ]
  }
} 
`

**Reply by bluebill1049** [reply]
why not fork that codesandbox and add a button with setValue and see if that works.

**Reply by niralparmar** [reply]
@bluebill1049  setValue works with button but not "onChange" of react-select. however, once values is getting set, it messes up react-select display 
[codesandbox](https://codesandbox.io/s/musing-glitter-0brjd)

**Reply by niralparmar** [reply]
@bluebill1049  the way i have implemented "setValue" isnt that correct way to use it?

**Reply by bluebill1049** [reply]
That's not correct usage. you should leverage `onChange` function to return what the component's value. 

<Controller as={<Input />} onChange={() => 'test'} />

'test' will be assigned to input's value

i know this is not very clear, that's why we changed to render props in the next version v6.
